sdk/cosmos/azure-cosmos/azure/cosmos/_base.py

```diff
@@ -674,3 +676,53 @@ def validate_cache_staleness_value(max_integrated_cache_staleness):
     if max_integrated_cache_staleness < 0:
         raise ValueError("Parameter 'max_integrated_cache_staleness_in_ms' can only be an "
                          "integer greater than or equal to zero")
+
+
+def _stringify_auto_scale(offer):
+    auto_scale_params = None
+    max_throughput = offer.auto_scale_max_throughput
+    increment_percent = offer.auto_scale_increment_percent
+    if max_throughput is not None:
+        auto_scale_params = {"maxThroughput": max_throughput}
+        if increment_percent is not None:
+            auto_scale_params["autoUpgradePolicy"] = {"throughputPolicy": {"incrementPercent": increment_percent}}
+    auto_scale_settings = json.dumps(auto_scale_params)
+
+    return auto_scale_settings
+
+
+def _set_throughput_options(offer, options):
+    offer_throughput = offer
+    request_options = options
+    if offer_throughput is not None:
+        try:
+            if offer_throughput.auto_scale_max_throughput:
+                request_options['autoUpgradePolicy'] = _stringify_auto_scale(offer=offer_throughput)
+            elif offer_throughput.auto_scale_increment_percent:
+                raise ValueError("auto_scale_max_throughput must be supplied in "
+                                 "conjunction with auto_scale_increment_percent")
+            elif offer_throughput.offer_throughput:
+                request_options["offerThroughput"] = offer_throughput.offer_throughput
+
+        except AttributeError:
+            if isinstance(offer_throughput, int):
+                request_options["offerThroughput"] = offer_throughput
+
+
+def _deserialize_throughput(throughput):
+    throughput_properties = throughput
+    if 'offerAutopilotSettings' in throughput_properties[0]['content'] and 'autoUpgradePolicy' in \
+            throughput_properties[0]['content']['offerAutopilotSettings']:
+        return ThroughputProperties(properties=throughput_properties[0], auto_scale_max_throughput= \
+            throughput_properties[0]['content']['offerAutopilotSettings']['maxThroughput'],
+                                    auto_scale_increment_percent=throughput_properties[0]['content']
+                                    ['offerAutopilotSettings']['autoUpgradePolicy']['throughputPolicy'][
+                                        'incrementPercent'])
+
+    if 'offerAutopilotSettings' in throughput_properties[0]['content']:
+        return ThroughputProperties(properties=throughput_properties[0],
+                                    auto_scale_max_throughput=throughput_properties[0]['content'][
+                                        'offerAutopilotSettings']['maxThroughput'])
+
+    return ThroughputProperties(offer_throughput=throughput_properties[0]["content"]["offerThroughput"],
+                                properties=throughput_properties[0])
```

Comment:
I feel like this section could be simplified a bit with:
```python
def _deserialize_throughput(throughput):
    try:
        max_throughput = throughput_properties[0]['content']['offerAutopilotSettings']['maxThroughput']
    except (KeyError, TypeError):  #  Adding TypeError just in case one of these dicts is None
        max_throughput = None
    try:
        increment_percent = throughput_properties[0]['content']['offerAutopilotSettings']['autoUpgradePolicy']['throughputPolicy']['incrementPercent']
    except (KeyError, TypeError):
        increment_percent = None
    try:
        throughput = throughput_properties[0]["content"]["offerThroughput"]
    except (KeyError, TypeError):
        throughput = None
    return ThroughputProperties(
        auto_scale_max_throughput=max_throughput,
        auto_scale_increment_percent=increment_percent,
        offer_throughput=throughput,
        properties=throughput_properties[0]
    )
```